pkgs/development/misc/rappel/default.nix

```diff
@@ -0,0 +1,34 @@
+{ clang
+, fetchFromGitHub
+, libedit
+, stdenv
+}:
+
+stdenv.mkDerivation rec {
+  name = "rappel";
+
+  src = fetchFromGitHub {
+    owner = "yrp604";
+    repo = "rappel";
+    rev = "95a776f850cf6a7c21923a2100b605408ef038de";
+    sha256 = "0fmd15xa6hswh3x48av4g1sf6rncbiinbj7gbw1ffvqsbcfnsgcr";
+  };
+
+  buildInputs = [ libedit ];
+  nativeBuildInputs = [ clang ];
+
+  makeFlags = "CC=clang";
+
+  installPhase = ''
+    make
+    cp -r bin $out/
```

Comment:
Because `$out` doesn't already exist at this point, this puts the `rappel` executable into `$out/` instead of `$out/bin/`, which does not give you a working `rappel` in PATH (tested: nix 2.3 on NixOS master).

You can remove `make` (handled in `buildPhase`) and replace this with:

```
  installPhase = ''
    mkdir -p $out/bin
    cp bin/rappel $out/bin/
  '';
```

Reply:
alternatively:

```
installPhase = ''
  install -D -m755 bin/rappel $out/bin/rappel
'';
```

Reply:
it executable also needs `nasm`, so you could write it as:
```
  nativeBuildInputs = [ makeWrapper ];

  installPhase = ''
    runHook preInstall

    install -Dm755 -t $out/bin bin/rappel
    wrapProgram $out/bin/rappel --prefix PATH : "${nasm}/bin"

    runHook postInstall
  '';
```